src/arp.c

```diff
@@ -132,16 +132,19 @@ void arp_seq_init(arp_seq_t* s) {
 }
 
 bool arp_seq_set_state(arp_seq_t *s, arp_seq_state state) {
+#ifndef RUNNING_TESTS
```

Comment:
curious, why is this needed? does it need to be done in other places that use `irqs_pause/resume`? if so, would it be better to move `ifndef` to the actual functions?

Reply:
I just found [this](https://github.com/monome/libavr32/pull/44#issuecomment-458303142) conversation which leads me to believe that this `#ifndef` isn't needed on the latest master. This was how I made the tests pass when I first made this branch, I can clean this up.

Reply:
ah right, this rings a bell. i think either way is fine - `irqs` stuff is not needed for running tests anyway.

Reply:
Removed `RUNNING_TESTS`, realized there was actually an error testing `json_read_object_cached`, added a make target so that if you have valgrind installed you can run `make leaks`, and found some leaks in the test code - nothing actually affecting the JSON encoder itself I think.